processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -97,30 +113,51 @@ func (mtp *metricsTransformProcessor) transform(md pdata.Metrics) pdata.Metrics
 	return pdatautil.MetricsFromMetricsData(mds)
 }
 
-// update updates the original metric content in the metric pointer.
-func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
-	// metric name update
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform mtpTransform) {
 	if transform.NewName != "" {
 		metric.MetricDescriptor.Name = transform.NewName
 	}
 
 	for _, op := range transform.Operations {
-		// update label
-		if op.Action == UpdateLabel {
+		switch op.configOperation.Action {
+		case UpdateLabel:
 			mtp.updateLabelOp(metric, op)
+		case AggregateLabels:
+			mtp.aggregateLabelsOp(metric, op)
+		case AggregateLabelValues:
+			mtp.aggregateLabelValuesOp(metric, op)
 		}
 	}
 }
 
-func (mtp *metricsTransformProcessor) updateLabelOp(metric *metricspb.Metric, op Operation) {
-	for _, label := range metric.MetricDescriptor.LabelKeys {
-		if label.Key != op.Label {
-			continue
-		}
-		// label key update
-		if op.NewLabel != "" {
-			label.Key = op.NewLabel
+// getLabelIdxs gets the indices of the labelSet labels' indices in the metric's descriptor's labels field
+// Returns the indices slice and a slice of the actual labels selected by this slice of indices
+func (mtp *metricsTransformProcessor) getLabelIdxs(metric *metricspb.Metric, labelSet map[string]bool) ([]int, []*metricspb.LabelKey) {
+	labelIdxs := make([]int, 0)
+	labels := make([]*metricspb.LabelKey, 0)
+	for idx, label := range metric.MetricDescriptor.LabelKeys {
+		_, ok := labelSet[label.Key]
+		if ok {
+			labelIdxs = append(labelIdxs, idx)
+			labels = append(labels, label)
 		}
-		// label value update
 	}
+	return labelIdxs, labels
+}
+
+// maxInt64 returns the max between num1 and num2
+func (mtp *metricsTransformProcessor) maxInt64(num1 int64, num2 int64) int64 {
```

Comment:
Note you can write this as:
```suggestion
func (mtp *metricsTransformProcessor) maxInt64(num1, num2 int64) int64 {
```